cli/cli/helpers/engine_manager/engine_manager.go

```diff
@@ -352,19 +353,14 @@ func (manager *EngineManager) startEngineWithGuarantor(ctx context.Context, curr
 	}
 	hostMachinePortBinding := engineGuarantor.getPostVisitingHostMachineIpAndPort()
 
-	engineClient, clientCloseFunc, err := getEngineClientFromHostMachineIpAndPort(hostMachinePortBinding)
```

Comment:
`grpc.Dial` does not build the connection until it is first used so we technically can call it before the engine port is forwarded but it is cleaner to move this logic after.